src/server/config/config.js

```diff
@@ -61,13 +61,25 @@ export class Config {
     if (_.isPlainObject(key)) {
       config = override(config, key);
```

Comment:
@tylersmalley 
- Question : should I replace `config = override(config, key);` by `config = override(config, _replaceEnvVars(key));` ?

Reply:
I'm not entirely sure what it's used for, maybe testing.  @spalger do you know?

Reply:
Not sure it is used. Gut feeling is it was just an "expected" behavior of a method like this.

Reply:
So, should I replace override(config, key) by override(config, _replaceEnvVars(key)) ?

Reply:
My remcommendation from https://github.com/elastic/kibana/pull/16988#issuecomment-379050543 is to move all of this into https://github.com/elastic/kibana/blob/master/src/cli/serve/read_yaml_config.js and do it once when reading `kibana.yml` from the file system

Reply:
@spalger 
OK, I will do so.
Thanks for this first review.